### DISCUSSION TITLE
Wrap the entire app

### DISCUSSION THREAD

**Discussion by ofrades** [TanStack/router#853] [Ideas]
react-router-dom has the [BrowserRouter](https://reactrouter.com/en/main/router-components/browser-router) used to wrap the entire application. It enables the use of the routing features in the child components.

When using auth0 providers for example we need to wrap it inside a router.

How do we achieve the same using tanstack/react-router?

more info: https://github.com/remix-run/react-router/issues/9324#issuecomment-1268309880

**Comment by tannerlinsley** [comment]
You technically do the same thing and use a layout root at the root and place all of your routes inside of that… but that’s going to create some noise in the route ID namespace. I think I would rather add an InnerWrap option or just allow for composition inside of the RouterProvider with <Matches> something. Thoughts?

**Reply by ofrades** [reply]
I have tried to add the provider to the `new RootRoute({ component: Auth0Provider });` and it works :tada:

**Reply by tannerlinsley** [reply]
To not dirty up your route paths and ids, try using the InnerWrap option on the router or RouterProvider
